queue/Queue.test.js

```diff
@@ -6,4 +6,19 @@ describe('Queue tests', () => {
     [1, 2, 3, 4, 5].forEach(q.enqueue, q);
     expect(q.dequeue()).toBe(1);
   });
+
+  test('invoking dequeue on empty Queue throws EmptyQueueException', () => {
+    const q = new Queue();
+    try {
+      q.dequeue();
+    } catch (err) {
+      expect(err.message).toBe('dequeue called on empty Queue');
+    }
+  });
```

Comment:
Maybe instead of using the try/catch block, because it will not hit the expect and fail if no exception is thrown, change the execution of dequeue to use this syntax.

`expect(() => q.dequeue()).toThrowError(EmptyQueueException);`